docs/design/connection-tester.md

```diff
@@ -0,0 +1,50 @@
+# Connection Tester
+
+The goal of this document is to sketch out possible approaches for implementing _Connection testing_ in the context of iPaaS. This backend concept is introduced to support the use case to quickly test whether a given `Connection`  (which is the combination of a `Connector` and a `ConnectorConfiguration`) is actually usable. The typical use case is, that the user pushes a button "Test connection" on the "Connection Details" page which actually triggers the test of the connection and that can be either successful or failing. When it fails, a detailed error message describes not only the error itself but also how to resolve the error (e.g. "Cannot connect because of invalid credentials"). 
+
+## Communication flow
+
+When the user triggers this test function from within the UI, a backend call to `ipaas-rest` will be issued, which in turn will trigger a (still to be defined) method on the referenced connector with the concrete configuration given. This method, let's call it `ping()`, must be side effect free, which should be possible for most endpoints. The method takes the configuration as argument for the connector and returns a result objects which contains the result of the check (ok / nok) and in case of a failure the error message (question: Should we already think about I18N here ?)
```

Comment:
@rhuss yeah so a connector will age, and when 2 or 3 years down the line the citizen developer starts using this connector it may no longer work as expected as let's say the twitter API slightly changed. So it's not only a thing to 'monitor', there should also be a test at creation time to test that the proposed solution is working; i.e. the return data format can have slightly changed.

Yes is happens before further integration but my point is that on a grand scheme we *know* what the message input and output is. Our plumbing can likely we abstract this message as some payload String; or "Connector + user specific ConnectorConfiguration", I'm sure we can send in an actual test message. I'm just saying this should be there in addition to what you are proposing. I'm happy to skip on it for this sprint, but I think we will need to have it before going out. We need to get this right, and we need to make it easy to build and validate each step as the citizen developer goes.